pyriemann/embedding.py

```diff
@@ -438,28 +443,209 @@ def locally_linear_embedding(
+        Metric for the gradient descent.
+    verbosity : int, default=0
+        Level of information printed by the optimizer while it operates:
+        0 is silent, 2 is most verbose.
+    max_iter : int, default=10_000
+        Maximum number of iterations used for the gradient descent.
+    max_time : int default=300
+        Maximum time on the run time of the gradient descent in seconds.
+    random_state : int, default=None
+        Pass an int for reproducible output across multiple function calls.
+
+    Attributes
+    ----------
+    embedding_ : ndarray, shape (n_matrices, n_components, n_components)
+        Embedding matrices of the training set.
+
+    Notes
+    -----
+    .. versionadded:: 0.9
+
+    References
+    ----------
+    .. [1] `Geometry-Aware visualization of high dimensional Symmetric
+        Positive Definite matrices
+        <https://openreview.net/pdf?id=DYCSRf3vby>`_
+        T. de Surrel, S. Chevallier, F. Lotte and F. Yger.
+        Transactions on Machine Learning Research, 2025
+    """
+
+    def __init__(
+        self,
+        n_components=2,
+        perplexity=None,
+        metric="riemann",
+        verbosity=0,
+        max_iter=10000,
+        max_time=300,
+        random_state=None,
+    ):
+        self.n_components = n_components
+        self.perplexity = perplexity
+        self.metric = metric
+        self.verbosity = verbosity
+        self.max_iter = max_iter
+        self.max_time = max_time
+        self.random_state = random_state
+
+    def _compute_similarities(self, X):
+        """Compute the high-dimensional symmetrized conditional similarities
+        p_{ij} for t-SNE.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of high-dimensional SPD matrices to reduce.
+
+        Returns
+        -------
+        P : ndarray, shape (n_matrices, n_matrices)
+            Symmetrized conditional probabilities of X.
+        """
+        n_matrices, _, _ = X.shape
+        Dsq = pairwise_distance(X, metric=self.metric, squared=True)
+        Dsq = Dsq.astype(np.float32, copy=False)
+        # Use _binary_search_perplexity from sklearn to compute conditional
+        # probabilities such that they approximately match the desired
+        # perplexity
+        conditional_P = _binary_search_perplexity(Dsq, self.perplexity, 0)
+
+        # Symmetrize the conditional probabilities
+        P = conditional_P + conditional_P.T
+        return P / (2 * n_matrices)
+
+    def _compute_low_affinities(self, Y):
+        """Compute the low-dimensional similarities q_{ij} for t-SNE.
+
+        Parameters
+        ----------
+        Y : ndarray, shape (n_matrices, n_components, n_components)
+            Set of low-dimensional SPD matrices.
+
+        Returns
+        -------
+        Q : ndarray, shape (n_matrices, n_matrices)
+            Low-dimensional similarities conditional probabilities of Y.
```

Comment:
Can you be more precise on descriptions?
For `Q`, are these similarities or probabilities? Are they conditional or joint?